### DISCUSSION TITLE
DS18x20 is not supported on GPIO16

### DISCUSSION THREAD

**Discussion by brenndorfler** [arendst/Tasmota#14799] [Q&A] [answered]
Hi, 
in the writeup of the sensor modules (https://tasmota.github.io/docs/DS18x20/) it is written that GPIO16 is not supported for DS18x20.
What is the reason for this?
If I'm right, the special on GPIO16 is the use for wakeup from Deep Sleep.
And during the boot the pin is on High-level.
 
High during boot should not be irrelevant for DS18x20.
I also do not use Deep Sleep.

Can I still not use GPIO16?
According to my tests it works. 
Are there any side effects?

Thanks!

Translated with www.DeepL.com/Translator (free version)

**Comment by sfromis** [comment]
IIRC, the reason is less expectation of reliably getting it to work on gpio16. Just did a small test, D1 Mini with a DS18B20 working on gpio12, but not on gpio16. And when users have reported problems getting DS18B20 to work on gpio16, the cure has been to use another pin.

**Comment by Jason2866** [comment]
Not supported does not mean it is not working. Since GPIO16 is a restricted GPIO there is no time invested to find unpredictable strange behaviours. You can use, if it works for you. If it fails, you walk alone.

**Comment by brenndorfler** [comment]
Okay, I have two Wemos D1 boards. 
7x DS18B20 sensors are connected to the 1st board. The bus is about 10 meters of twisted pair, shielded, telephone cable. Two encapsulated sensors with about one meter cable are connected to the bus in three places and another single sensor of the same type is connected on 4th place. I use current tasmota-DE 10.1.0 version (and already tried some others).
I connected the bus to D0 (GPIO16) with 4k7 PullUp. The installation runs without problems and is stable, although the signal shows a very bad slew rate when monitored with a oscilloscope.
To the 2nd D1 board three DS18S20 have been connected (currently @ D2/GPIO4). The onewire-bus has a similar length, the leads to the sensors are very short (10 cm). After switching on the power supply, all sensors are recognized and deliver a plausible measurement result each second. However, at some point the measurement results freezes on by another sensor and there is no change in the measurement value. 
When I inspect the bus with an oscilloscope, I see a similar signals as with the 1st Wemos. However, I also see that bidirectional communication is taking place, i.e. both the ESP and all three DS18S20s are sending recognizable telegrams (Ping-Pong). 
I have already exchanged one DS18S20 with a DS18B20. It feels like it takes longer for the sensors to freeze now. I suspect that the conversion in the DS18S20 chip does not work properly (W1_CONVERT_TEMP), for whatever reason. 
Question: Does it make sense to incomment the line in void Ds18x20Convert(void)?:
//delay(750); // 750ms should be enough for 12bit conv. 
If I restart the board (soft reboot), it doesn't help, the values stay frozen. If I disconnect the power supply of the DS18S20 it works again. I didn't read anything in the datasheet that the DS18S20 somehow disconnects itself from the bus!?!
I now have the following ideas:
Either the 3.3V power supply from the Wemos D1 is not sufficient for the DS18S20, but seven sensors are connected to the other board without any problems. I have already exchanged the Wemos D1 with each other. No change. I don’t really believe this…
Maybe it is a thermal effect in the DS18x20 (triggered by the measurements every second perhaps?).
The onebus ran stable with all 10 sensors on one bus (mixed topology). Host was a DS9097U (with DS2480) but I wanted to get away from the usb-hardware, because my home automation runs in a VM and passing the hardware is sometimes a problem. Difference between before and now, I think is 3,3V now instead of 5V before and an active pull-up vs. a passive pull-up now, which of course has a positive effect on the slew rate of the signal.
A question of understanding: 
If I set weblog 4, I should see a sensor loss in debug, right? At this time there no warnings or something else…
How can I increase the measurement interval to e.g. 10 seconds [Ds18x20EverySecond()] as a test?
Regards
  brenndorfler

**Reply by sfromis** [reply]
You may have gotten bad clones. https://github.com/cpetrich/counterfeit_DS18B20

If you want to play with the driver and longer intervals, you could have a counter in the "every second" function, and exit unless it is at your chosen interval.

**Comment by lalo-uy** [comment]
To improve communication you can lower the pullup, as low as 820 ohms.
But in the long run, bad clones stuck. I have long lines with 8 and 10
sensors on raspberry pi, and the only solution was use a gpio to power down
and up the sensors.

**Comment by brenndorfler** [comment, marked as answer]
Hi all, 
I finally replaced the old sensors with new DS18B20. 
Now it runs stable, even on D0 (GPIO16) for more than 36 hours yet.
Looks like I really had a problem with pure sensors.
I would not have thought of that, because they were running more or less stable for 7 years on a DS9097U USB-Adapter in an even "wilder" setup before.
Thanks to all for the support.
